File: main/java/com/noahliu/nestedrecyclerdemo/MainActivity.java
package com.noahliu.nestedrecyclerdemo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    MyAdapter myAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        List<MyData> data = makeData();
        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        myAdapter = new MyAdapter(data, new MyAdapter.OnItemClick() {
            @Override
            public void onItemClick(MyData.NestedData data, MyData myData) {
                Toast.makeText(MainActivity.this
                        , "chosen"+myData.getTitle()+"of "+data.getNesTitle()
                        , Toast.LENGTH_SHORT).show();
            }
        });
        recyclerView.setAdapter(myAdapter);

    }

    private List<MyData> makeData(){
        List<MyData> data = new ArrayList<>();
        String[] title = new String[]{"Monday","Tuesday","Wednesday","Thursday","Friday"};
        String[] nesTitle = new String[]{"Braised beef", "Tomato scrambled eggs",
                                         "Fried chicken drumsticks", "Sweet and sour fish", "Baked stewed rice",
                                         "Fried shrimps", "Braised salmon"};

        for (int i = 0; i < title.length; i++) {
            int r = (int)(Math.random()*7);
            List<MyData.NestedData> nesArray = new ArrayList<>();
            for (int j = 0; j < r+1; j++) {
                nesArray.add(new MyData.NestedData(nesTitle[(int)(Math.random()*7)]));
            }
            data.add(new MyData(title[i],nesArray));
        }
        return data;
    }
}

File: main/java/com/noahliu/nestedrecyclerdemo/MainActivity.kt
package com.example.f

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView


class MainActivity : AppCompatActivity() {
    private var myAdapter: MyAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val data: List<MyData> = makeData()
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        myAdapter = MyAdapter(data, object : MyAdapter.OnItemClick {
            override fun onItemClick(data: MyData.NestedData?, myData: MyData?) {
                if (myData != null) {
                    if (data != null) {
                        Toast.makeText(
                            this@MainActivity,
                            "chosen" + myData.title + "of " + data.nesTitle,
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                }
            }
        })
        recyclerView.adapter = myAdapter
    }

    private fun makeData(): List<MyData> {
        val data: MutableList<MyData> = ArrayList()
        val title = arrayOf("Monday", "Tuesday", "Wednesday", "Thursday", "Friday")
        val nesTitle = arrayOf(
            "Braised beef", "Tomato scrambled eggs",
            "Fried chicken drumsticks", "Sweet and sour fish", "Baked stewed rice",
            "Fried shrimps", "Braised salmon"
        )
        for (i in title.indices) {
            val r = (Math.random() * 7).toInt()
            val nesArray: MutableList<MyData.NestedData> = ArrayList()
            for (j in 0 until r + 1) {
                nesArray.add(MyData.NestedData(nesTitle[(Math.random() * 7).toInt()]))
            }
            data.add(MyData(title[i], nesArray))
        }
        return data
    }
}
